pkg/monitor/monitor.go

```diff
@@ -177,6 +183,12 @@ func (m *Monitor) driverMonitor() {
 							continue
 						}
 
+						// delete volume attachments if the node is down for this pod
+						err = m.cleanupVolumeAttachments(&pod)
```

Comment:
This will delete all the pods in the cluster. Needs be inside the if condition below which checks if the pod is on the node where PX is offline.

Reply:
Actually, this should be outside the loop for the pods. If the storage driver is offline we should go through all the volumeattachments and delete the ones that are attached to that node. So even if someone deletes a pod manually from a node where the storage is offline this will clean it up.

Reply:
Makes sense - will change

Reply:
Changed to two different functions-
1. For podMonitor - `cleanupVolumeAttachmentsByPod`, which deletes all VolumeAttachment for a given pod.
2. For driverMonitor - `cleanupVolumeAttachmentsByNode`, which deletes all VolumeAttachments for a given node.

Reply:
I ran this lastest code on my cluster and I noticed that the VolumeAttachments weren't continuously being deleted/recreated on every `driverMonitor` loop as they were before.

So, this fix handles both cases (driverMonitor and podMonitor)